engine/src/main/java/org/terasology/utilities/collection/TypeMap.java

```diff
@@ -70,7 +71,13 @@ public boolean containsValue(T value) {
         }
         return null;
     }
-
+    public <U extends T> U get(Class<U> key, Function<? super Class<? extends T>, ? extends T> ifAbsentPut) {
```

Comment:
I didn't check the use case here, so I might be wrong: have you thought about doing it the other way round? [Map.getOrDefault(Object key, V defaultValue)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#getOrDefault-java.lang.Object-V-) could be useful.